src/lib/list/selection-list.ts

```diff
@@ -178,6 +178,15 @@ export class MatListOption extends _MatListOptionMixinBase
   }
 
   ngOnDestroy(): void {
+    if (this.selected) {
+      // We have to delay this until the next tick in order
+      // to avoid changed after checked errors.
+      Promise.resolve().then(() => {
+        this._setSelected(false);
+        this.selectionList._reportValueChange();
```

Comment:
How about moving that into a single line and just doing `this.selected = false`?

Reply:
That feels a little side-effect-ey since we're basically using a property as a method.

Reply:
Why is that? I think the setter's main intention is to update the internal state and reflect it within in the model. Otherwise the setter wouldn't be needed at all.

Reply:
True, although I think of it mostly if you're setting it from the outside. That being said it does avoid having to duplicate some of the logic. I've switched it.